flows/contact.go

```diff
@@ -11,17 +11,32 @@ import (
 
 // Contact represents a single contact
 type Contact struct {
-	uuid     ContactUUID
-	name     string
-	language utils.Language
-	timezone *time.Location
-	urns     URNList
-	groups   GroupList
-	fields   *Fields
+	uuid              ContactUUID
+	name              string
+	language          utils.Language
+	backdownLanguages utils.LanguageList
+	timezone          *time.Location
+	urns              URNList
+	groups            GroupList
+	fields            *Fields
 }
 
+// SetLanguage sets the language for this contact
 func (c *Contact) SetLanguage(lang utils.Language) { c.language = lang }
-func (c *Contact) Language() utils.Language        { return c.language }
+
+// Language gets the language for this contact
+func (c *Contact) Language() utils.Language { return c.language }
+
+// PreferredLanguages gets all languages for this contact in order of preference
+func (c *Contact) PreferredLanguages() utils.LanguageList {
+	var languages utils.LanguageList
+
+	if c.language != utils.NilLanguage {
+		languages = append(languages, c.language)
+	}
+
+	return append(languages, c.backdownLanguages...)
+}
```

Comment:
Come to think of it, RapidPro shouldn't remove duplicates from backdown. If a user has language 'eng' and backdown 'eng', we should provide both of those to goflow, because even if the contact language changes, we still need to know that 'eng' is a valid backdown language.

So I think...
  * contact.language and contact.backdownLanguages are allowed to overlap
  * contact.PreferredLanguages should return with duplicates removed
  * run.SetLanguages shouldn't add flow language if it's already included